File: VICSApp/CoreSystem/DHCD/FormUI/Vote/ElectionVoteList.vb
Public Class ElectionVoteList

    Private Sub ToolStripButton4_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripButton4.Click
        filldgv()
    End Sub

    Public Sub filldgv()
        Dim t As New DataTable
        Try
            Dim delecode As Integer = 0
            Dim eleccode As Integer = 0
            Dim candicode As Integer = 0
            Try
                eleccode = ToolStripTextBox1.Text
            Catch ex As Exception
                eleccode = 0
            End Try
            Try
                delecode = ToolStripTextBox2.Text
            Catch ex As Exception
                delecode = 0
            End Try

            Try
                candicode = ToolStripTextBox3.Text
            Catch ex As Exception
                candicode = 0
            End Try

            t = Mainform.BenlyDal.ElectionVotes_getlist(Mainform.workingmeeting, eleccode, delecode, candicode)

        Catch ex As Exception
            MsgBox("Lỗi :" + ex.Message)
        End Try
        DataGridView1.DataSource = t
        ToolStripStatusLabel2.Text = DataGridView1.RowCount
        Dim totalright As Integer = 0
        For Each dr As DataRow In t.Rows
            totalright = totalright + dr.Item("Votes")
        Next

        ToolStripStatusLabel16.Text = Mainform.addthousandseperator(totalright.ToString)


    End Sub

    Private Sub ElectionVoteList_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        Me.MdiParent = Mainform
        filldgv()
    End Sub

    Private Sub ToolStripButton3_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripButton3.Click
        If DataGridView1.SelectedRows.Count = 0 Then
            MsgBox("Bạn phải chọn ít nhất một bản ghi")
            Exit Sub
        Else
            If MsgBox("Bạn có chắc chắn XÓA phiếu bầu của đại biểu : " + DataGridView1.CurrentRow.Cells("Delegatename").Value, MsgBoxStyle.OkCancel + MsgBoxStyle.Critical + MsgBoxStyle.ApplicationModal + MsgBoxStyle.DefaultButton2, "XÓA CỔ ĐÔNG") = MsgBoxResult.Ok Then
                Try
                    Mainform.BenlyDal.ElectionVotes_delete_all(Mainform.workingmeeting, DataGridView1.CurrentRow.Cells("Electioncode").Value, DataGridView1.CurrentRow.Cells("delegatecode").Value, DataGridView1.CurrentRow.Cells("Candidatecode").Value)
                Catch ex As Exception
                    MsgBox("Lỗi : " + ex.Message)
                End Try
            End If

        End If
        filldgv()
    End Sub

    Private Sub ToolStripButton1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripButton1.Click
        Dim f As New Electionvote_ins_update("Add", 0, 0)
        f.ShowDialog()
        filldgv()
    End Sub

    Private Sub ToolStripButton2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles ToolStripButton2.Click
        If DataGridView1.SelectedRows.Count = 0 Then
            MsgBox("Bạn phải chọn ít nhất một bản ghi")
            Exit Sub
        End If
        Dim f As New Electionvote_ins_update("Update", DataGridView1.CurrentRow.Cells("Electioncode").Value, DataGridView1.CurrentRow.Cells("delegatecode").Value)
        f.ShowDialog()
        filldgv()
    End Sub
End Class

File: VICSApp/CoreSystem/DHCD/FormUI/Vote/ElectionVoteList.cs
using System;
using System.Data;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace pmDHCD
{
    public partial class ElectionVoteList
    {
        public ElectionVoteList()
        {
            InitializeComponent();
            _ToolStripButton1.Name = "ToolStripButton1";
            _ToolStripButton2.Name = "ToolStripButton2";
            _ToolStripButton3.Name = "ToolStripButton3";
            _ToolStripButton4.Name = "ToolStripButton4";
        }

        private void ToolStripButton4_Click(object sender, EventArgs e)
        {
            filldgv();
        }

        public void filldgv()
        {
            var t = new DataTable();
            try
            {
                int delecode = 0;
                int eleccode = 0;
                int candicode = 0;
                try
                {
                    eleccode = Conversions.ToInteger(ToolStripTextBox1.Text);
                }
                catch (Exception ex)
                {
                    eleccode = 0;
                }

                try
                {
                    delecode = Conversions.ToInteger(ToolStripTextBox2.Text);
                }
                catch (Exception ex)
                {
                    delecode = 0;
                }

                try
                {
                    candicode = Conversions.ToInteger(ToolStripTextBox3.Text);
                }
                catch (Exception ex)
                {
                    candicode = 0;
                }

                t = My.MyProject.Forms.Mainform.BenlyDal.ElectionVotes_getlist(My.MyProject.Forms.Mainform.workingmeeting, eleccode, delecode, candicode);
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Lỗi :" + ex.Message);
            }

            DataGridView1.DataSource = t;
            ToolStripStatusLabel2.Text = DataGridView1.RowCount.ToString();
            int totalright = 0;
            foreach (DataRow dr in t.Rows)
                totalright = Conversions.ToInteger(Operators.AddObject(totalright, dr["Votes"]));
            ToolStripStatusLabel16.Text = My.MyProject.Forms.Mainform.addthousandseperator(totalright.ToString());
        }

        private void ElectionVoteList_Load(object sender, EventArgs e)
        {
            MdiParent = My.MyProject.Forms.Mainform;
            filldgv();
        }

        private void ToolStripButton3_Click(object sender, EventArgs e)
        {
            if (DataGridView1.SelectedRows.Count == 0)
            {
                Interaction.MsgBox("Bạn phải chọn ít nhất một bản ghi");
                return;
            }
            else if (Interaction.MsgBox(Operators.AddObject("Bạn có chắc chắn XÓA phiếu bầu của đại biểu : ", DataGridView1.CurrentRow.Cells["Delegatename"].Value), (MsgBoxStyle)((int)MsgBoxStyle.OkCancel + (int)MsgBoxStyle.Critical + (int)MsgBoxStyle.ApplicationModal + (int)MsgBoxStyle.DefaultButton2), "XÓA CỔ ĐÔNG") == MsgBoxResult.Ok)
            {
                try
                {
                    My.MyProject.Forms.Mainform.BenlyDal.ElectionVotes_delete_all(My.MyProject.Forms.Mainform.workingmeeting, Conversions.ToDecimal(DataGridView1.CurrentRow.Cells["Electioncode"].Value), Conversions.ToDecimal(DataGridView1.CurrentRow.Cells["delegatecode"].Value), Conversions.ToDecimal(DataGridView1.CurrentRow.Cells["Candidatecode"].Value));
                }
                catch (Exception ex)
                {
                    Interaction.MsgBox("Lỗi : " + ex.Message);
                }
            }

            filldgv();
        }

        private void ToolStripButton1_Click(object sender, EventArgs e)
        {
            var f = new Electionvote_ins_update("Add", 0, 0);
            f.ShowDialog();
            filldgv();
        }

        private void ToolStripButton2_Click(object sender, EventArgs e)
        {
            if (DataGridView1.SelectedRows.Count == 0)
            {
                Interaction.MsgBox("Bạn phải chọn ít nhất một bản ghi");
                return;
            }

            var f = new Electionvote_ins_update("Update", Conversions.ToInteger(DataGridView1.CurrentRow.Cells["Electioncode"].Value), Conversions.ToInteger(DataGridView1.CurrentRow.Cells["delegatecode"].Value));
            f.ShowDialog();
            filldgv();
        }
    }
}